includes/multi-currency/Utils.php

```diff
@@ -13,6 +13,28 @@
  * Class that controls Multi-Currency Utils.
  */
 class Utils {
+
+	/**
+	 * The Store API route patterns that should be handled by the WooPay session handler.
+	 */
+	const STORE_API_ROUTE_PATTERNS = [
+		'@^\/wc\/store(\/v[\d]+)?\/cart$@',
+		'@^\/wc\/store(\/v[\d]+)?\/cart\/add-item$@',
+		'@^\/wc\/store(\/v[\d]+)?\/cart\/remove-item$@',
+		'@^\/wc\/store(\/v[\d]+)?\/cart\/apply-coupon$@',
+		'@^\/wc\/store(\/v[\d]+)?\/cart\/remove-coupon$@',
+		'@^\/wc\/store(\/v[\d]+)?\/cart\/select-shipping-rate$@',
+		'@^\/wc\/store(\/v[\d]+)?\/cart\/update-customer$@',
+		'@^\/wc\/store(\/v[\d]+)?\/cart\/update-item$@',
+		'@^\/wc\/store(\/v[\d]+)?\/cart\/extensions$@',
+		'@^\/wc\/store(\/v[\d]+)?\/checkout\/(?P<id>[\d]+)@',
+		'@^\/wc\/store(\/v[\d]+)?\/checkout$@',
+		'@^\/wc\/store(\/v[\d]+)?\/order\/(?P<id>[\d]+)@',
+		// The route below is not a Store API route. However, this REST endpoint is used by WooPay to indirectly reach the Store API.
+		// By adding it to this list, we're able to identify the user and load the correct session for this route.
+		'@^\/wc\/v3\/woopay\/session$@',
+	];
```

Comment:
Even though unlikely, this list may change over time and we may end up needing to maintain it in two places. I'm considering whether it would make more sense to pass this through the `gateway_context`, as this is more related to WooPay and WooPay is a feature of WooPayments.

Reply:
I changed this in [b99e4d2](https://github.com/Automattic/woocommerce-payments/pull/9302/commits/b99e4d22df36e7253a782e690f8195923d955657).

I'm using now [a more comprehensive method](https://github.com/woocommerce/woocommerce/blame/013cef959e2ef663a9b0837bac6388dfc086f58c/plugins/woocommerce/includes/class-woocommerce.php#L532-L538) from Core, but since it's just 3 months old, I'm reimplementing it for MCCY.

The major problem with that was `@^\/wc\/v3\/woopay\/session$@`, which IMO shouldn't really be part of that constant, but as far as I could test it, it seems that this pattern is not important for multi-currency.

I can add a product with a different currency to the cart and checkout with WooPay as usual. Not sure if there would be something else to test?

This was introduced for MCCY in #8816, and before that it used to be just `if ( $is_frontend_request )`. The testing instructions from #8816 tested well for me.

Reply:
This seems like a great move, sourcing from core is the best scenario we can have 👍